GoogleSignIn/Sources/GIDProfileData_Private.h

```diff
@@ -26,7 +34,12 @@ NS_ASSUME_NONNULL_BEGIN
                          name:(NSString *)name
                     givenName:(nullable NSString *)givenName
                    familyName:(nullable NSString *)familyName
-                     imageURL:(nullable NSURL *)imageURL;
+                     imageURL:(nullable NSURL *)imageURL NS_DESIGNATED_INITIALIZER;
+
+/// Initialize with id token.
+- (nullable instancetype)initWithIDToken:(OIDIDToken *)idToken;
```

Comment:
I think this is a better place.

The reason to put it here is that `GIDSignIn.m` will use this initializer as well. It would be not reasonable to invoke a method in  `GIDProfileDataFetcher` to generate a `GIDProfileData` object.

Reply:
The profile data items in the ID token and UserInfo endpoint response share keys so having the same class handle decoding both seems simpler. 

> The reason to put it here is that `GIDSignIn.m` will use this initializer as well.

We could add a synchronous method to to API to fetch GIDProfileData from only the ID token.

Reply:
Done.